File: testutils-logrecorder/src/main/java/org/testit/testutils/logrecorder/api/LogEntry.java
package org.testit.testutils.logrecorder.api;

import lombok.Data;
import lombok.NonNull;


/**
 * Represents a single log entry. Includes the following data:
 * <ul>
 * <li>the name of the logger ({@link #loggerName})</li>
 * <li>the level on which the entry was logged ({@link #level})</li>
 * <li>the (formatted) message ({@link #message})</li>
 * </ul>
 *
 * @see LogLevel
 */
@Data
public class LogEntry {
    @NonNull
    private final String loggerName;
    @NonNull
    private final LogLevel level;
    @NonNull
    private final String message;
}

File: logrecorder/logrecorder-api/src/main/kotlin/de/novatec/logrecorder/api/LogEntry.kt
package de.novatec.logrecorder.api


/**
 * Contains the data of a single entry of the recorded log.
 *
 * Includes the following data:
 *
 * - the name of the logger ([logger])
 * - the [level][LogLevel] on which the entry was logged ([level])
 * - the (formatted) message ([message])
 *
 * @since 1.0
 * @see LogLevel
 */
data class LogEntry(
    val logger: String,
    val level: LogLevel,
    val message: String
)
